src/components/Menu/Menu.tsx

```diff
@@ -65,29 +72,24 @@ export const Menu = ({ className, ...other }: MenuProps) => {
 const MenuButton = ({ children, className, ...other }: MenuButtonProps) => {
   const buttonClassNames = clsx(styles['menu__button'], className);
   return (
-    <HeadlessMenu.Button
-      as={Button}
-      className={buttonClassNames}
-      status="neutral"
-      {...other}
-    >
-      <>
+    <HeadlessMenu.Button as={React.Fragment}>
+      <Button className={buttonClassNames} status="neutral" {...other}>
```

Comment:
This is how Headless recommends it in their [docs](https://headlessui.com/react/menu#rendering-a-different-element-for-a-component) (second code block).

Throws type error if not done this way now since it no longer merges the `as` types